### PR TITLE
Add naive chooseDouble

### PR DESCRIPTION
**Pull request by charleso**
@jystic 

If you have a second, this is probably wrong.

### CODE REVIEW

File: core/src/main/scala/hedgehog/Gen.scala [hunk 1]
```diff
@@ -163,14 +163,27 @@ object Gen extends GenImplicits2 {
...
     integral[M, Long](Range.constant(lo.toLong, hi.toLong)).map(_.toChar)
 
+  /**********************************************************************/
+  // Combinators - Fractional
+
+  def double[M[_]](range: Range[Double])(implicit F: Monad[M]): Gen[M, Double] =
+    Gen((size, seed) => {
+      val (x, y) = range.bounds(size)
+      val (s2, a) = seed.chooseDouble(x, y)
+      a.point[Gen[M, ?]].run(size, s2)
```
Comment (jystic) [inline core/src/main/scala/hedgehog/Gen.scala]: This looks like it's not going to shrink?

Comment (charleso) [inline core/src/main/scala/hedgehog/Gen.scala]: > This looks like it's not going to shrink?

Whoops. Fixed. Look ok now?

Comment (jystic) [inline core/src/main/scala/hedgehog/Gen.scala]: 👍

**Comment by jystic** [comment]
👍 I can't say anything about the mersenne twister algorithm for getting double, but the rest looks legit

**Pull request merged**